src/classic/clvm_tools/stages/stage_2/optimize.rs

```diff
@@ -590,23 +627,30 @@ fn apply_null_optimizer<'a>(
 
 struct OptimizerRunner<'a> {
     pub name: String,
-    to_run: &'a dyn Fn(&mut Allocator, NodePtr, Rc<dyn TRunProgram>) -> Result<NodePtr, EvalErr>,
+    to_run: &'a dyn Fn(
+        &mut Allocator,
+        Rc<RefCell<HashMap<String, NodePtr>>>,
+        NodePtr,
+        Rc<dyn TRunProgram>,
+    ) -> Result<NodePtr, EvalErr>,
 }
 
 impl<'a> OptimizerRunner<'a> {
     pub fn invoke(
         &self,
         allocator: &mut Allocator,
+        optimized: Rc<RefCell<HashMap<String, NodePtr>>>,
```

Comment:
is this the same as what's called `memo` in other parts of the code? I think both might warrant a comment somewhere explaining what it's for.

Reply:
ya, i renamed it